application/usecases/instagram_posts.go

```diff
@@ -0,0 +1,164 @@
+package usecases
+
+import (
+	"TML_TBot/config"
+	"TML_TBot/domain/models"
+	"bytes"
+	"fmt"
+	"image"
+	"image/png"
+	"net/http"
+	"os"
+	"reflect"
+
+	"github.com/Davincible/goinsta/v3"
+)
+
+type InstagramPostsController struct {
+	feedItems []FeedItem
+	codes     CodesSet
+}
+
+type FeedItem struct {
+	text  string
+	image *[]byte
+}
+
+// Use a set to avoid duplicate codes
+type CodesSet map[string]void
+
+type void struct{}
+
+const INSTAGRAM_CACHE_FILE = "./.cache/lastInstagramPosts.json"
+
+func NewInstagramPostsController() *InstagramPostsController {
+	var codes CodesSet
+	err := readJSONFileToStruct(&codes, INSTAGRAM_CACHE_FILE)
+	if err != nil {
+		config.Log.Info("Cached posts not found. Starting from ground...")
+	}
+
+	c := &InstagramPostsController{codes: codes}
+	return c
+}
+
+func (t *InstagramPostsController) Run() ([]models.TGMessage, error) {
+
+	var insta *goinsta.Instagram
+	insta, err := goinsta.Import("./.cache/.goinsta")
+	if err != nil {
+		fmt.Println("Cookies not found. Login in with user/password...")
+		insta = goinsta.New(os.Getenv("ig_user"), os.Getenv("ig_pass"))
+		err := insta.Login()
+		if err != nil {
+			fmt.Println(err)
```

Comment:
Let's use the application specific logger in all print calls (config.Log.Error if the error doesn't break the next lines, config.Log.Fatal if they do)

Reply:
Done